### ISSUE TITLE
False statement about methods (misleading recomendation)

### ISSUE THREAD

**Issue by cohadar** [golang/tour#97]
Context: https://tour.golang.org/methods/8

This statement from the tour is **false**:
"In general, all methods on a given type should have either value or pointer receivers, but not a mixture of both. (We'll see why over the next few pages.) "

It is perfectly valid (and desirable) to mix value and pointer receiver methods on a same type.
You use pointer for mutating methods and value for pure function methods.

On the next few pages it is hinted that mixed methods on a same type are bad because if interface has methods that are mixed than a value type does not implement that interface.
This is irrelevant because:
1. pointer to a type DOES implement mixed interfaces.
2. value type DOES NOT implement interfaces where ANY method needs a pointer.

In other words, pointer type has no detriment from mixed interfaces,
and value type cannot implement pointer interfaces even if they are unmixed.

**Comment by Raghav2211** [comment]
> Context: https://tour.golang.org/methods/8
> 
> This statement from the tour is **false**:
> "In general, all methods on a given type should have either value or pointer receivers, but not a mixture of both. (We'll see why over the next few pages.) "
> 
> It is perfectly valid (and desirable) to mix value and pointer receiver methods on a same type.
> You use pointer for mutating methods and value for pure function methods.
> 
> On the next few pages it is hinted that mixed methods on a same type are bad because if interface has methods that are mixed than a value type does not implement that interface.
> This is irrelevant because:
> 
> 1. pointer to a type DOES implement mixed interfaces.
> 2. value type DOES NOT implement interfaces where ANY method needs a pointer.
> 
> In other words, pointer type has no detriment from mixed interfaces,
> and value type cannot implement pointer interfaces even if they are unmixed.

👍

**Comment by theonlypwner** [comment]
As a concrete example,

```go
package main

type I interface {
	A()
	B()
}

type A struct{}

func (a A) A()  {}
func (a *A) B() {}

func main() {
	var a *A
	var i I = a
	_ = i
}
```

This is valid because of [this part of the spec](https://golang.org/ref/spec#Interface_types):
> The method set of any other type `T` consists of all methods declared with receiver type `T`. The method set of the corresponding pointer type `*T` is the set of all methods declared with receiver `*T` or `T` (that is, it also contains the method set of `T`).
